pinot-spi/src/main/java/org/apache/pinot/spi/data/DateTimeFormatSpec.java

```diff
@@ -199,6 +270,26 @@ public static void validateFormat(String format) {
               "Incorrect format type: %s in format: %s. Must be of '[0-9]+:<TimeUnit>:SIMPLE_DATE_FORMAT:pattern'",
               formatTokens[FORMAT_TIMEFORMAT_POSITION], format);
     }
+    return formatTokens;
+  }
+
+  /**
+   * Validates the pipe format string in the dateTimeFieldSpec
+   */
+  public static String[] validatePipeFormat(String format) {
+    Preconditions.checkNotNull(format, "Format string in dateTimeFieldSpec must not be null");
+    String[] formatTokens = StringUtils.split(format, PIPE_SEPARATOR, MAX_FORMAT_TOKENS_PIPE);
+    Preconditions.checkState(formatTokens.length >= MIN_FORMAT_TOKENS_PIPE
+            && formatTokens.length <= MAX_FORMAT_TOKENS_PIPE,
+        "Incorrect format: %s. Must be of the format 'EPOCH|<timeUnit>(|<size>)'"
+            + " or 'SDF|<timeFormat>(|<timezone>)' or 'TIMESTAMP'");
+    if (formatTokens.length == MAX_FORMAT_TOKENS_PIPE
+        && formatTokens[FORMAT_SIZE_POSITION].equals(TimeFormat.EPOCH.toString())) {
```

Comment:
Any reason why we are not validating the `SDF` format here?

Reply:
Added